cloud.ganttproject.colloboque/src/main/kotlin/cloud/ganttproject/colloboque/DevServer.kt

```diff
@@ -59,5 +71,53 @@ class ColloboqueHttpServer(port: Int, private val colloboqueServer: ColloboqueSe
       "/" -> newFixedLengthResponse("Hello")
       else -> newFixedLengthResponse(Status.NOT_FOUND, MIME_PLAINTEXT, "Not Found")
     }
+}
+
+class ColloboqueWebSocketServer(port: Int, private val colloboqueServer: ColloboqueServer) :
+  NanoWSD("localhost", port) {
+  override fun openWebSocket(handshake: IHTTPSession?): WebSocket {
+    return WebSocketImpl(handshake)
+  }
+
+  private class WebSocketImpl(handshake: IHTTPSession?) : WebSocket(handshake) {
+    private fun parseInputXlog(message: String): InputXlog? = try {
+      Json.decodeFromString<InputXlog>(message)
+    } catch (e: Exception) {
+      LOG.error("Failed to parse $message")
```

Comment:
pls use placeholders for logging the input which comes from the outside (remember the recent log4j issues?) and _log the exception too_, it carries information which is important for debugging
```suggestion
      LOG.error("Failed to parse {}", message, e)
```

Reply:
Thanks! Fixed